module/userExtension/helper.js

```diff
@@ -211,304 +212,373 @@ module.exports = class UserExtensionHelper {
-              }
-            } else {
-              let updateQuery = {};
+              if (userRole.programOperation && userRole.programs) {
+                for (const program of userRole.programs) {
+                  const programObjectId = programIdMap[program];
 
-              if (userRole.entityOperation) {
-                let userRoleToUpdate;
+                  if (!programObjectId) {
+                    userRole.status = messageConstants.apiResponses.PROGRAM_NOT_FOUND;
+                    userRolesUploadedData.push(userRole);
+                    continue outerloop;
+                  }
 
-                if (existingUser.roles && existingUser.roles.length > 0) {
-                  userRoleToUpdate = _.findIndex(existingUser.roles, {
-                    code: userRole.role,
-                  });
-                }
+                  const roles = platform_role_array;
+                  let entry = userInformation.platformRoles.find(
+                    (pr) => pr.programId.toString() === programObjectId.toString()
+                  );
 
-                if (!(userRoleToUpdate >= 0)) {
-                  userRoleToUpdate = existingUser.roles.length;
-                  userRoleMap[userRole.role].entities = new Array();
-                  existingUser.roles.push(userRoleMap[userRole.role]);
-                }
+                  if (!entry) {
+                    entry = { programId: programObjectId, roles: [] };
+                    userInformation.platformRoles.push(entry);
+                  }
 
-                existingUser.roles[userRoleToUpdate].entities = existingUser.roles[userRoleToUpdate].entities.map(
-                  (eachEntity) => eachEntity.toString(),
-                );
-
-                if (userRole.entityOperation == 'OVERRIDE') {
-                  existingUser.roles[userRoleToUpdate].entities = [userRole.entity];
-                  existingUser.roles[userRoleToUpdate].acl = userRole.acl;
-                } else if (userRole.entityOperation == 'APPEND' || userRole.entityOperation == 'ADD') {
-                  existingUser.roles[userRoleToUpdate].entities.push(userRole.entity);
-                  existingUser.roles[userRoleToUpdate].entities = _.uniq(existingUser.roles[userRoleToUpdate].entities);
-                } else if (userRole.entityOperation == 'REMOVE') {
-                  _.pull(existingUser.roles[userRoleToUpdate].entities, userRole.entity);
-                  removeUserFromEntity = true;
+                  for (const role of roles) {
+                    if (!entry.roles.includes(role)) {
+                      entry.roles.push(role);
+                      // Emit create event for new role
+                      kafkaEventPayloads.push({
+                        userId: userProfile.id,
+                        username: userProfile.username,
+                        programId: programObjectId,
+                        role,
+                        eventType: 'create',
```

Comment:
@borkarsaish65 define create/update/delete common variable if they are not present already and please use them every where